# EvoCELL 2019

## Introduction

This syllabus covers material that Cat Munro, Jacob Musser, and Andi Hejnol will present at the 2019 EvoCELL workshop in place of [Casey Dunn](http://dunnlab.org).

These days will include morning workshops on general computing skills based on the book [Practical Computing for Biologists](http://www.sinauer.com/practical-computing-for-biologists.html). These will be taught be [Casey Dunn](http://dunnlab.org). Morning workshops will be followed by afternoon workshops on specific applications of bioinformatics.

A few general notes about the course structure:

- Many details of the course are specific to UNIX-like operating systems, which include Linux and Apple macOS.

- We will also use UNIX for local command line, programing, and data processing exercises. This means that it will not be possible to use Microsoft Windows alone for most course activities. If your primary laptop is Microsoft Windows, there are a couple options. We suggest that you borrow a Mac laptop for the course. Alternatively, if you prefer you can install Linux on a Virtual Machine and run it from within Windows.

- Course content, schedule, code, and other files will be distributed on [github](https://github.com). Course materials are available at [neptune2016](https://github.com/neptune2016).


## To do before you arrive


- Bring a copy of [Practical Computing for Biologists](http://www.sinauer.com/practical-computing-for-biologists.html) (abbreviated in this document as PCfB). Complete the reading for the first class (see Schedule below) prior to arrival. Also download an familiarize yourself with the [example files](http://practicalcomputing.org) provided at the [book website](http://practicalcomputing.org).


- Create a [github](https://github.com) account. If you use an academic email address, you may be eligible for free upgrades.

- Install the required software listed below.


## Required software

Please install the following tools on your laptop before you arrive. This will reduce network load during the course, and also allow us to get started right away.

### All operating systems

- [Atom](https://atom.io) text editor

- [RStudio](https://www.rstudio.com/products/rstudio/download/#download)

- [Docker](https://www.docker.com/get-started)

### Apple macOS

- macOS command line developer tools. To see if you have them installed, open /Applications/Utilities/Terminal and type `gcc` at the command line. If `gcc` runs, you are probably all set. If it asks you if you would like to install the Developer tools, follow the instructions to do so. This will install `git`.

### Windows

- [git](https://git-scm.com)


## Instructions

### Git

## Schedule

This syllabus focuses on the classes to be taught by Casey.

## Reading

The course activities will walk through material covered in the following reading.

- Haddock, SHD and CW Dunn (2011). Practical Computing for Biologists. [amazon](http://www.amazon.com/Practical-Computing-Biologists-Steven-Haddock/dp/0878933913/ref=sr_1_1) *Casey wrote this book with my colleague Steve Haddock as an introduction to general computing skills for biologists. If you are not already comfortable at the command line then you should get this book as a reference.*

- Whickham, H (2017). R for Data Science. http://r4ds.had.co.nz *This book is free online at the provided link. It is an excellent introduction to data analysis in R, and more broadly how to think about data structure and analysis. It presents a coherent introduction to the Tidyverse, a set of R packages for general data manipulation and analysis. Our R coding will follow conventions in this book.*


### Class 1 (Monday 3 June): Getting Set Up, The Unix Command Line, Regular Expressions

9:00 - 13:00 -  Practical - Setting up your computer, regular expressions, introduction to the unix command line

14:00 - 18:00 - The Unix command line

16:00 Coffee Break

18:00 - 19:30 - Jaime Huerta-Cepas talk


### Class 2 (Wednesday 5 June): Version Control, Text Wrangling, Command Line (continued)

9:00 - 13:00 - Version control with git

14:00 - 15:30 - Working with text

16:00 - 17:00 - Integrating tools and designing workflows

18:30 - 19:10 - Jennifer Rainsford

### Class 3 (Thursday 6 June): Intro to R

9:00 - 13:00 - Intro to R and tidyverse (Jacob)

14:00 - 15:30 - Principles of data visualization

18:00 - 19:30 - Andi Hejnol: Comparative methods in EvoDevo

### Class 4 (Friday 7 June): Phylogenetic Comparative Methods for Evolutionary Functional Genomics


9:00 - 13:00 - Phylogenetic comparative analyses in R (Andi)

14:00 - 15:30 - Amy Maxmen, Career in Science Journalism

16:00  Coffee break

16:30 - 17:00 - Recap

18:00 - 19:30 - Amy Podium 

Amy will also give a 1-2 hour workshop on this day


## Reading

> Haddock and Dunn 2011. [Practical Computing for Biologists](http://www.sinauer.com/practical-computing-for-biologists.html). Sinauer and Associates.

## Other resources
